editor/src/components/canvas/canvas-strategies/strategies/paste-metastrategy.tsx

```diff
@@ -0,0 +1,247 @@
+              // This is here to appease the TS type checker edge case
+              return []
+            }
+
+            const existingIDs = getAllUniqueUids(editor.projectContents).allIDs
+            const elementWithUniqueUID = fixUtopiaElement(
+              currentValue.element,
+              new Set(existingIDs),
+            ).value
+
+            const reparentTarget: StaticReparentTarget =
+              strategy === 'REPARENT_AS_ABSOLUTE'
+                ? {
+                    strategy: strategy,
+                    insertionPath: target.parentPath,
+                    intendedCoordinates: absolutePositionForPaste(
+                      target,
+                      currentValue.originalElementPath,
+                      {
+                        originalTargetMetadata:
+                          elementPasteWithMetadata.targetOriginalContextMetadata,
+                        currentMetadata: canvasState.startingMetadata,
+                        originalPathTrees:
+                          interactionSession.interactionData.targetOriginalPathTrees,
+                        currentPathTrees: canvasState.startingElementPathTree,
+                      },
+                      interactionSession.interactionData.canvasViewportCenter,
+                    ),
+                  }
+                : { strategy: strategy, insertionPath: target.parentPath }
+
+            const indexPosition =
+              target.type === 'sibling'
+                ? absolute(
+                    MetadataUtils.getIndexInParent(
+                      editor.jsxMetadata,
+                      editor.elementPathTree,
+                      target.siblingPath,
+                    ) + 1,
+                  )
+                : front()
+
+            const result = insertWithReparentStrategies(
+              editor,
+              elementPasteWithMetadata.targetOriginalContextMetadata,
+              interactionSession.interactionData.targetOriginalPathTrees,
+              reparentTarget,
+              {
+                elementPath: currentValue.originalElementPath,
+                pathToReparent: elementToReparent(elementWithUniqueUID, currentValue.importsToAdd),
+              },
+              indexPosition,
+              canvasState.builtInDependencies,
+            )
+
+            if (result == null) {
+              return []
+            }
+
+            return foldAndApplyCommandsInner(
+              editor,
+              [],
+              [
+                ...result.commands,
+                updateSelectedViews('always', [...editor.selectedViews, result.newPath]),
+              ],
+              commandLifecycle,
+            ).statePatches
+          }),
+        ]
+      })
+
+      if (commands.length === 0) {
+        return emptyStrategyApplicationResult
+      }
+
+      return strategyApplicationResult([updateSelectedViews('always', []), ...commands])
+    },
+  }
+}
+
+export const pasteMetaStrategy =
```

Comment:
In our model we use `MetaStrategy` to represent something that will return an array of strategies, so rather than this taking the `mode` and deciding which one to return, this should just return the array of both strategies (after of course first checking the interaction type)

Reply:
fixed in https://github.com/concrete-utopia/utopia/pull/3798/commits/3e21d9a526d1af2cc2d467f72aff0a38811d5366